packages/asc-ui/src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,138 @@
+import { Tab, TabProps } from './Tab'
+
+// For more information about the accessibility features of this code check out the following references:
+// - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Tab_Role
+// - https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-1/tabs.html
+
+const TabList = styled.div.attrs({
+  role: 'tablist',
+})``
+
+const TabButton = styled.button.attrs({
+  role: 'tab',
+})``
+
+const TabPanel = styled.div.attrs({
+  role: 'tabpanel',
+  tabIndex: 0,
+})``
+
+export interface Tabs {
+  /**
+   * A descriptive label for the tabs, required for accessibility.
+   */
+  label: string
+  /**
+   * The children for the tabs, only `Tab` components are allowed.
+   */
+  children: ReactElement<PropsWithChildren<TabProps>, typeof Tab>[]
+}
+
+enum FocusDirection {
+  Forward = 1,
+  Backward = -1,
+}
+
+function formatTabId(id: string) {
+  return `tab-${id}`
+}
+
+function formatPanelId(id: string) {
+  return `panel-${id}`
+}
+
+export const Tabs: React.FC<Tabs> = ({ label, children }) => {
+  const allTabs = children.map(({ props }) => props.id)
+  const firstTab = allTabs[0] ?? ''
+  const [selectedTab, setSelectedTab] = useState(firstTab)
+  const [focusableTab, setFocusableTab] = useState(firstTab)
+  const tabListRef = useRef<HTMLDivElement>(null)
+
+  function focusTab(id: string) {
+    if (!tabListRef.current) {
+      return
+    }
+
+    const tabId = formatTabId(id)
+    const tabElement = tabListRef.current.querySelector(`#${tabId}`)
+
+    if (tabElement && tabElement instanceof HTMLElement) {
+      tabElement.focus()
+    }
+  }
+
+  function focusTowards(direction: FocusDirection) {
+    if (allTabs.length === 0) {
+      return
+    }
+
+    const currentIndex = allTabs.indexOf(focusableTab)
+    let nextIndex = currentIndex !== -1 ? currentIndex + direction : 0
+
+    if (nextIndex >= allTabs.length) {
+      nextIndex = 0
+    } else if (nextIndex < 0) {
+      nextIndex = allTabs.length - 1
+    }
+
+    focusTab(allTabs[nextIndex])
+  }
+
+  const focusNextTab = () => focusTowards(FocusDirection.Forward)
+  const focusPreviousTab = () => focusTowards(FocusDirection.Backward)
+
+  function handleKeyDown(event: React.KeyboardEvent<HTMLDivElement>) {
+    if (event.key === 'ArrowRight') {
+      focusNextTab()
+    }
+
+    if (event.key === 'ArrowLeft') {
+      focusPreviousTab()
+    }
+  }
+
+  return (
+    <>
+      <TabList aria-label={label} onKeyDown={handleKeyDown} ref={tabListRef}>
+        {children.map(({ props }) => {
+          const isSelected = props.id === selectedTab
+          const isFocusable = props.id === focusableTab
+          const tabId = formatTabId(props.id)
+          const panelId = formatPanelId(props.id)
+
+          return (
+            <TabButton
+              key={props.id}
+              id={tabId}
+              aria-controls={panelId}
+              aria-selected={isSelected}
+              tabIndex={isFocusable ? 0 : -1}
```

Comment:
tabindex shouldn't change on focus, only on selection. See implementation [here](https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-2/tabs.html). The problem with changing tabindex on focus is that when you leave the tab list and return, the focus also returns to the last focussed tab, not the selected tab. This doesn't match with _When focus moves into the tab list, places focus on the active tab element_ from that same link.

Reply:
This behaviour seems to be broken again, the focus seems to return to the last focussed tab, not the selected tab. I can't quite figure out why this changed, though :)